lua/terrortown/menus/gamemode/roles.lua

```diff
@@ -36,13 +37,18 @@ function CLGAMEMODEMENU:InitializeVirtualMenus()
 
         counter = counter + 1
 
-        virtualSubmenus[counter] = tableCopy(rolesMenuBase)
-        virtualSubmenus[counter].title = roleData.name
-        virtualSubmenus[counter].icon = roleData.iconMaterial
-        virtualSubmenus[counter].roleData = roleData
-        virtualSubmenus[counter].iconBadge = roleData.builtin and builtinIcon
-        virtualSubmenus[counter].iconBadgeSize = 8
-        virtualSubmenus[counter].basemenu = self
+        local virtualSubmenu = tableCopy(rolesMenuBase)
+        virtualSubmenu.title = roleData.name
+        virtualSubmenu.icon = roleData.iconMaterial
+        virtualSubmenu.roleData = roleData
+        virtualSubmenu.iconBadge = roleData.builtin and builtinIcon
+        virtualSubmenu.iconBadgeSize = 8
+        virtualSubmenu.basemenu = self
+        -- make sure that the virtual submenu will be shown
+        function virtualSubmenu:ShouldShow()
+            return true
+        end
```

Comment:
can you explain why this had to be added? Because it wasn't necessary before.

Reply:
See my reply here: https://github.com/TTT-2/TTT2/pull/1702#discussion_r1915704387 The virtual submenus need to be marked as visible once created to behave correctly.